### ISSUE TITLE
Bug in Resnet.py

### ISSUE THREAD

**Issue by SHANK885** [MVIG-SJTU/AlphaPose#1006]
Hi, I was trying to make modifications to the existing AlphaPose/configs/coco/resnet/256x192_res50_lr1e-3_1x-simple.yaml config file and change the **RESNET** backbone from resnet50 to resnet18 or resnet34. But after modifying the  **NUM_LAYERS: 50** to **34** or **18**, the train.py throws an error like this.
![2022-03-10_13-17](https://user-images.githubusercontent.com/36559699/157613587-84805121-6ee0-46a9-b5c7-1009b9b9720f.png)

I tried to find the issue and found that the issue is in **ResNet** class at this point at line 147-152:
`if architecture == "resnet18" or architecture == 'resnet34':
            self.block = BasicBlock
        else:
            self.block = Bottleneck
        self.layers = layers[architecture]`

where if we use resnet18 or resnet34 is uses BasicBlock instead of Bottleneck and there is no dcn in BasicBlock definition but we pass it while calling it here at line 162-169:
`self.layer1 = self.make_layer(self.block, 64, self.layers[0], dcn=stage_dcn[0])
self.layer2 = self.make_layer(self.block, 128, self.layers[1], stride=2, dcn=stage_dcn[1])
self.layer3 = self.make_layer(self.block, 256, self.layers[2], stride=2, dcn=stage_dcn[2])
self.layer4 = self.make_layer(self.block, 512, self.layers[3], stride=2, dcn=stage_dcn[3])`

**Comment by saicomagic** [comment]
@SHANK885 I have the same problem as you. Have you found the solution? Thanks!

**Comment by SHANK885** [comment]
This is the modified resnet module:
`class ResNet(nn.Module):
    """ ResNet """

    def __init__(self, architecture, norm_layer=nn.BatchNorm2d, dcn=None, stage_with_dcn=(False, False, False, False)):
        super(ResNet, self).__init__()
        self._norm_layer = norm_layer
        self.architecture = architecture
        assert architecture in ["resnet18", "resnet34", "resnet50", "resnet101", 'resnet152']
        layers = {
            'resnet18': [2, 2, 2, 2],
            'resnet34': [3, 4, 6, 3],
            'resnet50': [3, 4, 6, 3],
            'resnet101': [3, 4, 23, 3],
            'resnet152': [3, 8, 36, 3],
        }
        self.inplanes = 64
        if architecture == "resnet18" or architecture == 'resnet34':
            self.block = BasicBlock
        else:
            self.block = Bottleneck
        self.layers = layers[architecture]

        self.conv1 = nn.Conv2d(3, 64, kernel_size=7,
                               stride=2, padding=3, bias=False)
        self.bn1 = norm_layer(64, eps=1e-5, momentum=0.1, affine=True)
        self.relu = nn.ReLU(inplace=True)
        self.maxpool = nn.MaxPool2d(kernel_size=3, stride=2, padding=1)

        stage_dcn = [dcn if with_dcn else None for with_dcn in stage_with_dcn]

        if architecture == "resnet18" or architecture == 'resnet34':
            self.layer1 = self.make_layer(self.block, 64, self.layers[0])
            self.layer2 = self.make_layer(self.block, 128, self.layers[1], stride=2)
            self.layer3 = self.make_layer(self.block, 256, self.layers[2], stride=2)
            self.layer4 = self.make_layer(self.block, 512, self.layers[3], stride=2)
        else:
            self.layer1 = self.make_layer(self.block, 64, self.layers[0], dcn=stage_dcn[0])
            self.layer2 = self.make_layer(self.block, 128, self.layers[1], stride=2, dcn=stage_dcn[1])
            self.layer3 = self.make_layer(self.block, 256, self.layers[2], stride=2, dcn=stage_dcn[2])
            self.layer4 = self.make_layer(self.block, 512, self.layers[3], stride=2, dcn=stage_dcn[3])

        ############################################################


    def forward(self, x):
        x = self.maxpool(self.relu(self.bn1(self.conv1(x))))  # 64 * h/4 * w/4
        x = self.layer1(x)  # 256 * h/4 * w/4
        x = self.layer2(x)  # 512 * h/8 * w/8
        x = self.layer3(x)  # 1024 * h/16 * w/16
        x = self.layer4(x)  # 2048 * h/32 * w/32
        return x

    def stages(self):
        return [self.layer1, self.layer2, self.layer3, self.layer4]

    def make_layer(self, block, planes, blocks, stride=1, dcn=None):
        downsample = None
        if stride != 1 or self.inplanes != planes * block.expansion:
            downsample = nn.Sequential(
                nn.Conv2d(self.inplanes, planes * block.expansion,
                          kernel_size=1, stride=stride, bias=False),
                self._norm_layer(planes * block.expansion),
            )

        layers = []

        ###############    Modification        ####################
        if self.architecture == "resnet18" or self.architecture == 'resnet34':
            layers.append(block(self.inplanes, planes, stride, downsample,
                                norm_layer=self._norm_layer))

            self.inplanes = planes * block.expansion
            for i in range(1, blocks):
                layers.append(block(self.inplanes, planes,
                                    norm_layer=self._norm_layer))

        else:
            layers.append(block(self.inplanes, planes, stride, downsample,
                                norm_layer=self._norm_layer, dcn=dcn))

            self.inplanes = planes * block.expansion

            for i in range(1, blocks):
                layers.append(block(self.inplanes, planes,
                                    norm_layer=self._norm_layer))
        ###########################################################

        return nn.Sequential(*layers)`

**Comment by Fang-Haoshu** [comment]
Thanks, merge in commit https://github.com/MVIG-SJTU/AlphaPose/commit/d022c118cc84f2fcc476fd43a54e4308fca5df34
